src/karma-webpack.js

```diff
@@ -121,23 +121,13 @@ function Plugin(
   const applyPlugins = compiler.compilers || [compiler];
 
   applyPlugins.forEach(function(compiler) {
-    if (compiler.hooks) {
-      compiler.hooks.thisCompilation.tap(this.plugin, (compilation, params) => {
-        compilation.dependencyFactories.set(
-          SingleEntryDependency,
-          params.normalModuleFactory
-        );
-      });
-      compiler.hooks.make.tapAsync(this.plugin, this.make.bind(this));
-    } else {
-      compiler.plugin('this-compilation', (compilation, params) => {
-        compilation.dependencyFactories.set(
-          SingleEntryDependency,
-          params.normalModuleFactory
-        );
-      });
-      compiler.plugin('make', this.make.bind(this));
-    }
+    compiler.hooks.thisCompilation.tap(this.plugin, (compilation, params) => {
```

Comment:
Could you try with using `compilation` instead of `thisCompilation`, the latter is an alias if I remember right and will be removed in a feature version of `webpack`. But it's not important to do this now, I just saw it, so in case it bugs for whatever reason don't bother :)